agent/agent.go

```diff
@@ -101,8 +112,35 @@ func DefaultConfig() (Config, error) {
 	}, nil
 }
 
+func DefaultAgentConfig() (Config, error) {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return Config{}, errors.Wrap(err, "could not read the hostname")
+	}
+
+	return Config{
+		InstanceName: hostname,
+		DiscoveryTTL: 15 * time.Second,
+		CheckerTTL:   10 * time.Second,
+	}, nil
+}
+
 // Start the Agent which includes the registration against Consul Agent
 func (a *Agent) Start() error {
+	var wg sync.WaitGroup
+
+	// This number must match the number threads attached to the WaitGroup object
+	wg.Add(1)
+	errs := make(chan error, 4)
+
+	go func(wg *sync.WaitGroup) {
```

Comment:
Do we want to run this in a go routine? For me, it looks like this actually should be something blocking. Nothing else should be executed unless consul is up and running.
That's why actually you are force to put that `sleep`.
I would be consul api has some method to check if the agent is running

Reply:
> @rtorrero Added some comments.
> I cannot understand why you were forced to fork the benchmark package. This looks something related to your environment, because I don't find any relationship between consul agent and this library

@stefanotorresi might be able to explain this better as he helped me address this, but in summary, they are requiring `k8s.io/client-go v11.0.0+incompatible` (see https://github.com/aquasecurity/bench-common/blob/main/go.mod) which is incompatible with `v0.16.9` which is what `consul` uses.

Reply:
can we fix that in aquasecurity's codebase?

Reply:
we're going to drop this dependency soon anyway

Reply:
@arbulu89 I don't think that would work: `agent.Start(ctx)` will keep running in the background and block until `cancel()` is called on `ctx`, it will not return once started so, if you don't put this in a goroutine, nothing else will ever be started until the consul agent is stopped explicitly.